.github/workflows/ci.yml

```diff
@@ -271,6 +271,28 @@ jobs:
             cargo check --manifest-path "$toml";
           done
 
+  misc:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: dtolnay/rust-toolchain@stable
+
+      - uses: r7kamura/rust-problem-matchers@2c2f1016021a7455a6b5b4bbae31145f3b3cd83a #v1.4.0
+
+      - uses: Swatinem/rust-cache@b8a6852b4f997182bdea832df3f9e153038b5191 # v2.6.0
+        with:
+          shared-key: stable-cache
+          save-if: false
+
+      - name: Compile all tools in misc
+        run: |
+          set -e;
+
+          for toml in misc/**/Cargo.toml; do
+            cargo check --manifest-path "$toml";
+          done
+
```

Comment:
We could publish it instead? You seem to have bumped the version regularly? If you set `publish = true`, it would be part of our regular CI.

Reply:
We could also move it to the examples directory? To me it is a kind of example.

It is such a thin layer around our production-ready code that I don't really see the problem with it being in `examples`.

Most examples are simply opinionated in a certain way (like logging etc). We can explain that more in the README as well?

Reply:
> If you set publish = true, it would be part of our regular CI.

That would not solve the issue for `misc/keygen`.

Do you see any benefit of publishing `libp2p-server` to `crates.io` beyond that being a trigger for our CI action?

> We could also move it to the examples directory? To me it is a kind of example.

That would as well not solve the issue for `misc/keygen`. I don't feel strongly about it, but I wouldn't expect either to be in `examples/`.

Reply:
> > We could also move it to the examples directory? To me it is a kind of example.
> 
> That would as well not solve the issue for `misc/keygen`.

What is the use of `misc/keygen`? I never quite understood why we have it.

Reply:
> > If you set publish = true, it would be part of our regular CI.
> 
> That would not solve the issue for `misc/keygen`.
> 
> Do you see any benefit of publishing `libp2p-server` to `crates.io` beyond that being a trigger for our CI action?

Users can then install it via `cargo install`. For example, one could publish an AUR package for it.

I'd just like to avoid our CI to grow more than it needs to. `misc/` is becoming a bigger annoyance by the day because all sorts of pieces go in there. By adding a dedicated CI job for it, we are even more legitimating its existence when in reality, it should actually go away.

Reply:
> > > We could also move it to the examples directory? To me it is a kind of example.
> > 
> > 
> > That would as well not solve the issue for `misc/keygen`.
> 
> What is the use of `misc/keygen`? I never quite understood why we have it.

Say you want to deploy a new IPFS bootstrap server. You need a stable identity. You can generate a keypair with `keygen` and store it to a file. The new IPFS bootstrap server can then read the keypair from disk on startup.

Reply:
> > > If you set publish = true, it would be part of our regular CI.
> > 
> > 
> > That would not solve the issue for `misc/keygen`.
> > Do you see any benefit of publishing `libp2p-server` to `crates.io` beyond that being a trigger for our CI action?
> 
> Users can then install it via `cargo install`. For example, one could publish an AUR package for it.

Good point. Works for me.



> I'd just like to avoid our CI to grow more than it needs to. `misc/` is becoming a bigger annoyance by the day because all sorts of pieces go in there. By adding a dedicated CI job for it, we are even more legitimating its existence when in reality, it should actually go away.

This still leaves us with `keygen` untested :/

Reply:
> > I'd just like to avoid our CI to grow more than it needs to. `misc/` is becoming a bigger annoyance by the day because all sorts of pieces go in there. By adding a dedicated CI job for it, we are even more legitimating its existence when in reality, it should actually go away.
> 
> This still leaves us with `keygen` untested :/

Delayed this decision, i.e. marked `libp2p2p-server` as `publish = true` and removed the `misc/` action step for now.

https://github.com/libp2p/rust-libp2p/pull/4311/commits/2382f40446f2266c995f4305c5102d53ce50eaa4

https://github.com/libp2p/rust-libp2p/pull/4311/commits/a29d05b20bc1fa8480feceddb92c97966f2da335

Reply:
:+1: https://github.com/libp2p/rust-libp2p/issues/4370